lib/compress/zstd_ldm.h

```diff
@@ -28,18 +28,17 @@ extern "C" {
  * ZSTD_ldm_generateSequences():
  *
  * Generates the sequences using the long distance match finder.
- * The sequences completely parse a prefix of the source, but leave off the last
- * literals. Returns the number of sequences generated into `sequences`. The
- * user must have called ZSTD_window_update() for all of the input they have,
- * even if they pass it to ZSTD_ldm_generateSequences() in chunks.
+ * Generates at most `*nbSeqPtr` sequences into `sequences`, which parse a
+ * prefix of the source, and returns the number of unparsed literals at the end,
```

Comment:
small comment on return value : 
`returns the number of unparsed literals at the end`

This return value felt correct at a time when result of the "ldm" generator was passed to block compressor : it was more or less the amount of data the block compressor would have to work with.

Here I read it as : this is the place when "ldm" parsing will resume.

If that's correct, it feels more logical to return the amount of data processed from `src` instead.

What do you think ?

Reply:
I think I will just remove the return value for now, and fix it up later, in a patch where I only add the dynamic sequence resizing. The intention was to allow dynamically resizing the sequences storage, so it can resume parsing. However, this return value isn't enough to do that.

There are two "markers", the "nextToUpdate" position, where the LDM stopped inserting literals, and the number of trailing literals. We need to resume parsing at the nextToUpdate, even if there are more trailing literals, and also know how many extra literals to prepend to the first sequence.

Reply:
minor follow up : 
if return value becomes available, you might seize the opportunity to transform `size_t* nbSeqPtr ` from read/write argument,
into a more common `size_t nbSeq` read-only input argument, 
and provide resulting `nbSeq` as function return value.

sidenote : if this function cannot fail, it's probably good to mention it in documentation too.